File: lib/data/models/food.dart
class Food {
  final String id;
  final String donorId;
  final String foodName;
  final String quantity;
  final String description;
  final List<String> imageUrls;
  final String location;
  final DateTime expiryTime;
  final String status;
  final DateTime createdAt;
  final double? distanceM;

  Food({
    required this.id,
    required this.donorId,
    required this.foodName,
    required this.quantity,
    required this.description,
    required this.imageUrls,
    required this.location,
    required this.expiryTime,
    required this.status,
    required this.createdAt,
    this.distanceM,
  });

  factory Food.fromJson(Map<String, dynamic> json) {
    return Food(
      id: json['id'] as String,
      donorId: json['donor_id'] as String? ?? '',        // ← nullable
      foodName: json['food_name'] as String? ?? '',      // ← nullable
      quantity: json['quantity'] as String? ?? '',       // ← nullable
      description: json['description'] as String? ?? '', // ← nullable
      imageUrls: List<String>.from(json['image_urls'] ?? []),
      location: json['location'] as String? ?? '',       // ← nullable
      expiryTime: DateTime.parse(
          json['expiry_time'] as String? ?? DateTime.now().toIso8601String()),
      status: json['status'] as String? ?? 'available', // ← nullable
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : DateTime.now(),                              // ← nullable
      distanceM: (json['distance_m'] as num?)?.toDouble(),
    );
  }
}

File: backend/models/Food.js
import mongoose from "mongoose";

const foodSchema = new mongoose.Schema(
    {
        foodName: {
            type: String,
            required: true
        },

        quantity: {
            type: String,
            required: true
        },

        description: {
            type: String
        },

        preparedTime: {
            type: String
        },

        expiryTime: {
            type: String
        },

        location: {
            type: {
                type: String,
                enum: ["Point"],
                default: "Point"
            },

            coordinates: {
                type: [Number],
                required: true
            }
        },

        images: [
            {
                type: String
            }
        ],

        donorId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true
        },

        status: {
            type: String,
            enum: ["available", "reserved", "picked"],
            default: "available"
        }

    },
    { timestamps: true }
);

foodSchema.index({ location: "2dsphere" });

const Food = mongoose.model("Food", foodSchema);

export default Food;